src/core/segment_allocator.cc

```diff
@@ -12,8 +12,8 @@ constexpr size_t kSegmentShift = MI_SEGMENT_SHIFT;
 namespace dfly {
 
 SegmentAllocator::SegmentAllocator(mi_heap_t* heap) : heap_(heap) {
-  // mimalloc uses 4MiB segments and we might need change this code if it changes.
-  constexpr size_t kSegLogSpan = 32 - kSegmentIdBits + 3;
+  // mimalloc uses 32MiB segments and we might need change this code if it changes.
+  constexpr size_t kSegLogSpan = 32 - kSegmentIdBits + 3 + 3;
```

Comment:
This should be right based on the mapping in `mimalloc/types.h`.

If I missed anything plz let me know

Reply:
I believe this logic for segment allocator is broken now.
Anyway can you please find how to go to 4Mib segments again 
https://github.com/microsoft/mimalloc/pull/753

Reply:
@adiholden  why do you think it's broken? Basically it's about how many bits we allocate for a segment id and the segment offset. at the end they should sum up to 32.